### PR TITLE
JAMES-2945 Make Sender case insensitive in SMTP

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/AbstractSenderAuthIdentifyVerificationRcptHook.java [hunk 1]
```diff
@@ -68,10 +66,10 @@ private boolean isAnonymous(MaybeSender maybeSender) {
     private boolean senderMatchSessionUser(MaybeSender maybeSender, SMTPSession session) {
         Preconditions.checkArgument(!maybeSender.isNullSender());
 
-        String authUser = session.getUser().toLowerCase(Locale.US);
+        String authUser = session.getUser();
         String username = getUser(maybeSender.get());
 
-        return username.equals(authUser);
+        return username.equalsIgnoreCase(authUser);
```
Comment (rouazana) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/AbstractSenderAuthIdentifyVerificationRcptHook.java]: I guess it's not the responsibility of the hook to know is the user repository is case sensitive or not. The user repository should be responsible to check if a username belongs to its database.

**Comment by Arsnael** [comment]
CI restart...
test this please

**Comment by Arsnael** [comment]
Was :green_apple:

File: server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SenderAuthIdentifyVerificationRcptHook.java [hunk 2]
```diff
@@ -77,4 +77,8 @@ protected String getUser(MailAddress mailAddress) {
         }
     }
 
+    @Override
+    protected boolean isSenderAllowed(String user, String sender) {
+        return user.equalsIgnoreCase(sender);
```
Comment (mbaechler) [inline server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/SenderAuthIdentifyVerificationRcptHook.java]: from the documentation of this method:
```
     * <p>Note that this method does <em>not</em> take locale into account, and
     * will result in unsatisfactory results for certain locales.  The
     * {@link java.text.Collator} class provides locale-sensitive comparison.
```

**Comment by blackheaven** [comment]
> Ok, so now I want a test to show how it fixes the problem I spotted

To be honest I struggle to find a test case.

**Comment by mbaechler** [comment]
Did you tried a Turkish `İ`?

**Comment by blackheaven** [comment]
I've have tried the french `ö`, but it occurs to block at IMAP parsing.

**Comment by mbaechler** [comment]
did you tried to change the JVM default Locale to Turkish?

**Comment by blackheaven** [comment]
I did with `Locale.setDefault` and I still get:

```
org.apache.james.utils.SMTPSendingException: Error upon step Sender: 501 5.1.7 Syntax error in sender address
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**